### PR TITLE
Modifying YouTube Video to use YouTube Data API

### PR DESCRIPTION
**Pull request by QuantumCD**
A video is now selected at random from a specified playlist instead of
being hard-coded. I have it set to a LoTRO playlist, however, it'd be
advisable to set this to something that pertains to the site (no idea
what that'd be).

### CODE REVIEW

**Comment by QuantumCD** [comment]
Performance-wise, it looks like the API request is about 3 KB. Not bad for 10 videos. You can shrink this if you want, but the videos won't be as varied.

**Comment by rioforce** [comment]
So, what if I wanted to add a video that's not in a playlist? Can you make it where I can put either a playlist or a single video?

**Comment by le717** [comment]
That too is my concern. Can you update it to work something like 

``` javascript
if (ytVideo) {
    // Redefine video.id to equal ytVideo to reduce redundant code
} else { 
    // Use playlist
}
```

Off-topic: For some reason, Travis CI is not wanting to run my grunt setup, so I'm having to check everything manually...

File: js/script.js [hunk 1]
```diff
@@ -11,24 +11,48 @@
...
+
+    var video;
+
+    $.getJSON(apiString, function(data)
+    {
+        // Get a random video
+        video = data.data.items[randomVideoIndex].video;
+
+        // If something went bad...
+        if (video == null)
```
Comment (le717) [inline js/script.js]: Always use `===` and `!==` when doing comparisons like this. :wink:

File: index.html [hunk 2]
```diff
@@ -100,6 +98,6 @@
     <!-- Load the FeedEk library and our settings asynchronously
         http://css-tricks.com/thinking-async/ -->
     <script async src="js/FeedEk.js"></script>
-    <script defer src="js/script.js"></script>
+    <script async src="js/script.js"></script>
```
Comment (le717) [inline index.html:101]: Ah, so that was the Chrome bug. Now why didn't catch that? :+1:

File: js/script.js [hunk 3]
```diff
@@ -7,28 +7,62 @@
...
+
+    var randomVideoIndex = Math.floor((Math.random() * videoRequestCount));
+    var queryString = "?v=2&alt=jsonc&max-results=" + videoRequestCount;
+
+    var apiString = apiBaseUrl + playlistId + queryString;
+    console.log("Fetching YouTube data from: " + apiString);
+
+    var video;
+
+    if (ytVideoId == null)
```
Comment (le717) [inline js/script.js]: You still need to change this to `if (ytVideoId === null)`.

File: js/script.js [hunk 4]
```diff
@@ -7,28 +7,62 @@
...
+    {    
+        $.getJSON(apiString, function(data)
+        {
+            // Get a random video
+            video = data.data.items[randomVideoIndex].video;
+
+            // If something went bad...
+            if (video == null)
```
Comment (le717) [inline js/script.js]: As well as this to `if (video === null)`.

**Comment by QuantumCD** [comment]
Why?

**Comment by le717** [comment]
Your braces are confusing me a bit. It seems the playlist/video priority is
- If video and playlist is assigned, display video
- If video is not defined but video is, use playlist
- If playlist is not defined but video is, use video
- If none are defined, display error image

Is this correct?

**Comment by le717** [comment]
> Why?

[http://www.impressivewebs.com/why-use-triple-equals-javascipt/](http://www.impressivewebs.com/why-use-triple-equals-javascipt/)

That's why.

**Comment by QuantumCD** [comment]
Actually, the '==' will also be true if the variable is undefined. That's why I left it that way.

**Comment by le717** [comment]
So I see. Well, there are two options:
1. Change the `==` to simply `!`, removing the errors (see [http://labs.codecademy.com/BtMn#:workspace](http://labs.codecademy.com/BtMn#:workspace))
2. I merge it as-is.

**Comment by QuantumCD** [comment]
Wait, what "errors" are you referring to? That wasn't causing any errors. :P

**Comment by le717** [comment]
If you take the code with the `== null` and run it through [JSHint](http://jshint.com/) (the de-facto JavaScript lint checker), you'll see it reports the use of `==` as errors. In fact, all JS lint checkers report `==` as an error.

As for 

> Sorry, but this just doesn't look right to a C++ programmer. :P
> of course it doesn't. This is JavaScript, not C++. Though it has a lot of C-style syntax, it's not the exact same language.

@rioforce for final say on merge, as he did originally create this version of the Patcher.

**Comment by QuantumCD** [comment]
So you have merge rights, yet you want Rio to make the final decision? You need to reorganize your hierarchy. :P

**Comment by rioforce** [comment]
> So you have merge rights, yet you want Rio to make the final decision? You need to reorganize your hierarchy. :P

Actually, this was my project, and le717 just decided to re-write it, so yes, I wanted to approve it before he merged it.

**Pull request merged**